src/security_utilities_rust/src/microsoft_security_utilities_core/identifiable_secrets.rs

```diff
@@ -474,6 +556,40 @@ fn generate_base64_key_helper(checksum_seed: u64,
                                                             encode_for_url);
 }
 
+pub fn validate_common_annotated_key_signature(base64_encoded_signature: &str) -> Result<String, String> {
+    const REQUIRED_ENCODED_SIGNATURE_LENGTH: usize = 4;
+
+    if base64_encoded_signature.len() != REQUIRED_ENCODED_SIGNATURE_LENGTH {
+        return Err(format!("Base64-encoded signature {} must be 4 characters long.",
+                            base64_encoded_signature));
+    }
+
+    if base64_encoded_signature.chars().next().unwrap().is_digit(10) {
+        return Err(format!("The first character of the signature {} must not be a digit.",
+                            base64_encoded_signature));
+    }
+
+    for ch in base64_encoded_signature.chars() {
+        if !is_base62_encoding_char(ch) {
+            return Err(format!("Signature {} can only contain alphabetic or numeric values.",
```

Comment:
>Signature [](http://example.com/codeflow?start=32&length=9)

I'd love to see you embrace enquoting dynamic arguments using single quotes.

Signature 'BLAH' can only contain alphabetic, etc.

This is our constant pattern across our tooling. You may as well internalize it because I will harp on this NIT constantly. :)

Reply:
Will fix this. Won't leave it for later.